frame/mosaic-vault/src/lib.rs

```diff
@@ -0,0 +1,904 @@
+            
+			ensure!(min_fee < T::FeeFactor::get(), Error::<T>::MinFeeAboveFeeFactor);
+
+            <MinFee<T>>::put(min_fee);
+
+			Self::deposit_event(Event::MinFeeChanged{min_fee});
+
+			Ok(().into())
+		 }
+
+		 #[pallet::weight(10_000)]
+		 pub fn set_thresh_hold(origin: OriginFor<T>, new_fee_threshold: T::Balance) -> DispatchResultWithPostInfo {
+			 
+			T::AdminOrigin::ensure_origin(origin)?;
+
+			ensure!(new_fee_threshold < T::ThresholdFactor::get(), Error::<T>::ThresholdFeeAboveThresholdFactor);
+
+			<FeeThreshold<T>>::put(new_fee_threshold);
+
+			Self::deposit_event(Event::FeeThresholdChanged{new_fee_threshold});
+
+			Ok(().into())
+		 }
+
+		 #[pallet::weight(10_000)]
+		 pub fn deposit(
+			 origin: OriginFor<T>, 
+			 amount: T::Balance,
+			 asset_id: T::AssetId, 
+			 destination_address: T::AccountId, 
+			 remote_network_id: T::RemoteNetworkId,
+		 	 transfer_delay: T::TransferDelay,
+			) -> DispatchResultWithPostInfo {
+
+			let sender = ensure_signed(origin)?;
+
+			ensure!(Self::pause_status() == false, Error::<T>::ContractPaused);
+
+			ensure!(amount != T::Balance::zero(), Error::<T>::ZeroAmount);
+
+			Self::only_supported_remote_token(remote_network_id.clone(), asset_id.clone())?;
+
+			ensure!(Self::last_transfer(&sender).checked_add(Self::transfer_lockup_time()).ok_or(Error::<T>::Overflow)? < T::BlockTimestamp::now().as_secs(), Error::<T>::TransferNotPossible);
+           
+			ensure!(transfer_delay >= <MinTransferDelay<T>>::get(), Error::<T>::TransferDelayBelowMinimum);
+		
+			ensure!(transfer_delay <= <MaxTransferDelay<T>>::get(), Error::<T>::TransferDelayAboveMaximum);
+
+			ensure!(amount <= Self::max_asset_transfer_size(asset_id), Error::<T>::AmountAboveMaxAssetTransferSize);
+
+			ensure!(amount >= Self::min_asset_transfer_size(asset_id), Error::<T>::AmountBelowMinAssetTransferSize);
+			// 
+			let pallet_account_id = Self::account_id();            
+            // move funds to pallet amount
+		     T::Currency::burn_from(asset_id, &sender, amount).map_err(|_|Error::<T>::BurnFromFailed)?;
+
+			 Self::increase_total_value_transferred(asset_id, amount)?;
+
+			// update in_transfer_funds
+			let in_transfer_funds = Self::in_transfer_funds(asset_id);
+			let new_in_transfer_funds = in_transfer_funds.checked_add(&amount).ok_or(Error::<T>::Overflow)?;
+			<InTransferFunds<T>>::insert(asset_id, new_in_transfer_funds);
+           
+			<LastTransfer<T>>::insert(&sender, T::BlockTimestamp::now().as_secs());
+
+			let deposit_id = Self::generate_deposit_id(remote_network_id, &destination_address, pallet_account_id);
+            <Deposits<T>>::insert(asset_id, DepositInfo{asset_id, amount});
+
+			Self::deposit_event(Event::DepositCompleted{
+				sender,
+				asset_id,
+				remote_asset_id: Self::remote_asset_id(remote_network_id, asset_id),
+				remote_network_id,
+				destination_address,
+				amount, 
+				deposit_id,
+				transfer_delay
```

Comment:
used only in errors and event, which is informational only. either remove or use it. or commetn why not used

Reply:
thats right,  though i'm  trying to keep the events consistent with the deployed solidity contract , I guess the relayer expects these parameters, but ill ask to enure , removing will be ok